orchagent/txmonorch.cpp

```diff
@@ -0,0 +1,313 @@
+#include <linux/if_ether.h>
+
+#include <unordered_map>
+#include <utility>
+#include <exception>
+#include <string>
+
+#include "txmonorch.h"
+#include "orch.h"
+#include "port.h"
+#include "logger.h"
+#include "sai_serialize.h"
+#include "converter.h"
+#include "portsorch.h"
+#include <vector>
+
+extern PortsOrch*       gPortsOrch;
+
+//using namespace std::rel_ops;
+
+string tx_status_name [] = {"ok", "error", "unknown"};
+
+TxMonOrch::TxMonOrch(TableConnector appDbConnector,
+                     TableConnector confDbConnector,
+                     TableConnector stateDbConnector) :
+    Orch(confDbConnector.first, confDbConnector.second),
+    m_TxErrorTable(appDbConnector.first, appDbConnector.second),
+    m_stateTxErrorTable(stateDbConnector.first, stateDbConnector.second),
+    m_countersDb(COUNTERS_DB, DBConnector::DEFAULT_UNIXSOCKET, 0),
+    m_countersTable(&m_countersDb, COUNTERS_TABLE),
+    m_countersMapTable(&m_countersDb, COUNTERS_PORT_NAME_MAP),
+    m_pollTimer(new SelectableTimer(timespec { .tv_sec = 0, .tv_nsec = 0 })),
+    m_PortsTxErrStat(),
+    m_pollPeriod(0)
+{
+    auto executor = new ExecutableTimer(m_pollTimer, this, TXMONORCH_SEL_TIMER);
+    Orch::addExecutor(executor);
+
+    SWSS_LOG_NOTICE("TxMonOrch initialized with table %s %s %s\n",
+                    appDbConnector.second.c_str(),
+                    stateDbConnector.second.c_str(),
+                    confDbConnector.second.c_str());
+}
+
+void TxMonOrch::startTimer(uint32_t interval)
+{
+    SWSS_LOG_ENTER();
+
+    try
```

Comment:
not sure you need try catch here as i can't see what failure could happen in this scenario that will be caught by catch

Reply:
stop and start can throw exceptions.